metagen-lib-basics/src/main/java/io/virtdata/basicsmappers/from_long/to_string/AlphaNumericString.java

```diff
@@ -0,0 +1,45 @@
+package io.virtdata.basicsmappers.from_long.to_string;
+
+import java.util.function.LongFunction;
+
+import org.apache.commons.math3.random.MersenneTwister;
+
+import io.virtdata.api.ThreadSafeMapper;
+import io.virtdata.basicsmappers.from_long.to_long.Hash;
+
+@ThreadSafeMapper
+public class AlphaNumericString implements LongFunction<String>
+{
+    private final Hash hash = new Hash();
+    private final int length;
+    private final String availableChars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+    public AlphaNumericString(int length)
+    {
+        if (length < 0)
+        {
+            throw new RuntimeException("AlphaNumericString must have length >= 0");
+        }
+        this.length = length;
+    }
+
+    @Override
+    public String apply(long operand)
+    {
+        long randomSeed = hash.applyAsLong(operand);
+        MersenneTwister rnd = new MersenneTwister(randomSeed);
+        StringBuilder sb = new StringBuilder();
```

Comment:
For speed and low heap thrashing, it's generally a good idea to use a ThreadLocal around these StringBuilders, setting them to length zero for each call.